scripts/dialectic_cost_calculator.py

```
@@ -0,0 +1,439 @@
+"""
+
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+# Add project root to path for imports
+project_root = Path(__file__).parent.parent
+sys.path.insert(0, str(project_root))
+
+from rich.console import Console  # noqa: E402
```

Ruff (0.14.13)

Unused `noqa` directive (non-enabled: `E402`)

Remove unused `noqa` directive

(RUF100)